ecommerce/models.py

```diff
@@ -0,0 +1,277 @@
+"""Models for ecommerce"""
+from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+
+from mitxpro.models import AuditableModel, AuditModel, TimestampedModel
+
+
+class Product(TimestampedModel):
+    """
+    Representation of a purchasable product.
+    """
+
+    content_type = models.ForeignKey(
+        ContentType,
+        on_delete=models.PROTECT,
+        null=True,
+        help_text="content_object is a link to either a Course, CourseRun, or a Program",
```

Comment:
Can we/should we enforce the object types to be one of `Course`, `CourseRun`, or a `Program` if that's all we're supposed to allow?

Reply:
We can do that through a `clean` implementation similar to micromasters, but I'd like to leave it out for this PR to keep it small-ish.